go/ql/lib/semmle/go/dataflow/ExternalFlow.qll

```diff
@@ -213,16 +213,30 @@ predicate interpretModelForTest(QlBuiltins::ExtensionId madId, string model) {
   )
 }
 
+bindingset[p]
+private string cleanPackage(string p) {
+  exists(string noPrefix |
+    p = fixedVersionPrefix() + noPrefix
+    or
+    not p = fixedVersionPrefix() + any(string s) and
```

Comment:
Is this preferred to `not p.matches(fixedVersionPrefix() + "%")`?

Reply:
According to [this comment](https://github.com/github/codeql/pull/16941#discussion_r1677548036), @aschackmull thinks that `matches` would be fine but the approach above is the best version. In practice this is only being run on the set of package columns from MaD, so the numbers won't be too big.